File: backend/src/main/java/at/ac/tuwien/sepr/groupphase/backend/endpoint/dto/ConfigurationDto.java
package at.ac.tuwien.sepr.groupphase.backend.endpoint.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ConfigurationDto {
    private Long id;
    private Long clonedFromId;
    private Long calendarReferenceId;
    private String title;
    private String description;
    private List<RuleDto> rules;
    private boolean published;
}

File: frontend/src/app/dtos/configuration-dto.ts
export interface ConfigurationDto {
  id?: number,
  title: string,
  calendarReferenceId?: number,
  clonedFromId?: number,
  description: string,
  rules: RuleDto[],
  published: boolean,
  alreadyAdded?: boolean,
}

export interface PublicConfigurationDto extends ConfigurationDto {
  initialConfigurationId: number;
  alreadyCloned?: boolean;
  mine: boolean;
}

export enum MatchType {
  CONTAINS = 'CONTAINS',
  STARTS_WITH = 'STARTS_WITH',
  EQUALS = 'EQUALS',
  REGEX = 'REGEX'
}
export interface MatchDto {
  id?: number,
  summary?: string,
  description?: string,
  location?: string,
  summaryMatchType?: MatchType,
  descriptionMatchType?: MatchType,
  locationMatchType?: MatchType,
}

export enum EffectType {
  DELETE = 'DELETE',
  MODIFY = 'MODIFY',
  TAG = 'TAG'
}

export interface EffectDto {
  id?: number,
  changedTitle?: string,
  changedDescription?: string,
  location?: string,
  tag?: string,
  effectType?: EffectType
}

export interface RuleDto {
  id?: number,
  configId?: number,
  effect?: EffectDto,
  match?: MatchDto
}
